R/print-str.R

```diff
@@ -58,6 +104,23 @@ obj_print_footer.default <- function(x, ...) {
   invisible(x)
 }
 
+local_max_print <- function(max, frame = parent.frame()) {
+  max_print <- getOption("max.print")
+  if (is.null(max)) {
+    max <- max_print
+  }
+
+  stopifnot(is_integerish(max, 1L, finite = TRUE), max >= 0, max < 2147483648)
+  max <- as.integer(max)
+
+  if (max > max_print) {
+    # Avoid truncation in case we're forwarding to print()
+    local_options(max.print = max, .frame = frame)
+  }
+
+  structure(max, max_print = max_print)
```

Comment:
Should return a list.

Reply:
Why do we need both `max` and `max_print`? Could we take `max = getOption("max.print")` in the generic instead, to simplify things?